Issue 393 Notification Pattern (#1912)
iluwatar/java-design-patterns: notification/README.md

---
layout: pattern
title: Notification
folder: notification
permalink: /patterns/notification/
categories: Behavioural
language: en
tags:

> - Notification

Reviewer: For tags, I would put `Decoupling` and `Presentation`

Reviewer: Please remove the tag `Notification`

---
layout: pattern
title: Notification
folder: notification
permalink: /patterns/notification/
categories: Behavioural
language: en
tags:
- Notification
- Decoupling
- Presentation
---

## Intent

> A Notification is an object that collects information about notificationErrors during validation of data. When an notificationError appears the Notification is sent back to the view to display further information about the notificationErrors.

Reviewer: ```suggestion
Notification is an object that collects information about notification errors during the validation of data. When a notification error appears, the notification is sent back to the view to display further information about the notification errors.
```

## Explanation

Real world example

> There's a form to register for a course. We need to validate the inputs to make sure 

> > the inputs are valid.

Reviewer: Please use passive voice (see https://www.grammarly.com/blog/passive-voice/)

In plain words

> > An object that collects together information about notificationErrors and other information and communicates it to the presentation.

Reviewer: ```suggestion
> Notification is an object that collects together information about notification errors and communicates it to the presentation layer.
```
